tests/rptest/tests/franz_go_verifiable_test.py

```diff
@@ -83,3 +86,123 @@ def test_with_all_type_of_loads(self):
+            logger=self.logger,
+        )
+        self.s3client.create_bucket(self.s3_bucket_name)
+
+        self._node_for_franz_go = ctx.cluster.alloc(
+            ClusterSpec.simple_linux(1))
+
+        self._producer = FranzGoVerifiableProducer(ctx, self.redpanda,
+                                                   self.topic,
+                                                   SiWithFranzGoTest.MSG_SIZE,
+                                                   1000000000,
+                                                   self._node_for_franz_go)
+        self._seq_consumer = FranzGoVerifiableSeqConsumer(
+            ctx, self.redpanda, self.topic, SiWithFranzGoTest.MSG_SIZE,
+            self._node_for_franz_go)
+        self._rand_consumer = FranzGoVerifiableRandomConsumer(
+            ctx, self.redpanda, self.topic, SiWithFranzGoTest.MSG_SIZE, 10000,
+            20, self._node_for_franz_go)
+
+    def tearDown(self):
+        failed_deletions = self.s3client.empty_bucket(self.s3_bucket_name)
+        assert len(failed_deletions) == 0
+        self.s3client.delete_bucket(self.s3_bucket_name)
+
+    # In the future producer will signal about json creation
+    def _create_json_file(self):
+        small_producer = FranzGoVerifiableProducer(self.test_context,
+                                                   self.redpanda, self.topic,
+                                                   SiWithFranzGoTest.MSG_SIZE,
+                                                   10000,
+                                                   self._node_for_franz_go)
+        small_producer.start()
+        small_producer.wait()
+
+    @cluster(num_nodes=5)
+    def test_with_all_type_of_loads_and_si(self):
+
+        rpk = RpkTool(self.redpanda)
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.write', 'true')
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.read', 'true')
+        rpk.alter_topic_config(self.topic, 'retention.bytes',
+                               str(self.segment_size * 2))
+
+        # Need create json file for consumer at first
+        self._create_json_file()
+
+        # Verify that we really enabled shadow indexing correctly, such
+        # that some objects were written
+        objects = list(self.s3client.list_objects(self.s3_bucket_name))
+        assert len(objects) > 0
+        for o in objects:
+            self.logger.info(f"S3 object: {o.Key}, {o.ContentLength}")
+
+        self._producer.start()
+
+        # Produce some data
+        time.sleep(10)
+
+        self._seq_consumer.start()
+        self._rand_consumer.start()
+
+        for i in range(30):
+            node_for_restart = random.randrange(len(self.redpanda_nodes))
+            self.redpanda.restart_nodes(self.redpanda_nodes[node_for_restart],
+                                        start_timeout=900,
+                                        stop_timeout=900)
+            time.sleep(30)
```

Comment:
same here. why sleep? prefer testing for the condition you are interested in inside a wait_until block.

Reply:
In this instance the sleep is to wait for some throughput to happen (if we restart in a tight loop there might be very little time for messages to actually flow and get validated).

So maybe the wait condition could be to peek at the partition high water marks and wait for them to increase by some amount.  Or a walltime is kind of reasonable here too -- it depends if we want to define the "redpanda stays up and doesn't crash" in terms of message count or walltime.

Reply:
Yeh, we do something similar in other tests. For instance, start some producer and wait_until a related consumer has observed some threshold of fetched records. Same would apply to peeking at metrics etc... and ideally would then happen in most cases much quicker than 30 seconds.